src/main/java/edu/harvard/iq/dataverse/engine/command/impl/AssignRoleCommand.java

```diff
@@ -68,11 +68,22 @@ public RoleAssignment execute(CommandContext ctxt) throws CommandException {
                 throw new IllegalCommandException("User " + user.getUserIdentifier() + " is deactivated and cannot be given a role.", this);
             }
         }
+        if(isExistingRole(ctxt)){
+            throw new IllegalCommandException("User already has this role for this object", this);
```

Comment:
Can you please move this English text to src/main/java/propertyFiles/Bundle.properties?

That way it can be translated into other languages.

I guess "this object" is ok, but it's a bit computer-y. Maybe we could shorten to "User already hast this role." Or if you feel like it, you could try switching on DvObject and saying "collection", "dataset", or "file".

Reply:
@pdurbin due rebase issues i opened another PR ([here](https://github.com/IQSS/dataverse/pull/10474))